File: frontend/src/types/context.ts
/** GET /api/trade-sessions/{id}/context */
export interface ContextSummary {
  id: string;
  session_id: string;
  context_version: number;
  source_cutoff: string | null;
  is_stale: boolean;
  quality: string;
  payload: Record<string, unknown> | null;
  created_at: string;
}

File: backend/app/api/schemas/context.py
"""Pydantic schemas for Context Summary API (TP-1006)."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class ContextSummaryResponse(BaseModel):
    id: str
    session_id: str
    context_version: int
    source_cutoff: datetime | None
    is_stale: bool
    quality: str
    payload: dict[str, object] | None
    created_at: datetime
